src/unk_0203E880.c

```diff
@@ -605,90 +485,76 @@ u16 * FieldSystem_GetVarPointer (FieldSystem * fieldSystem, u16 varID)
+    return VarsFlags_CheckFlag(SaveData_GetVarsFlags(fieldSystem->saveData), flagID);
 }
 
-void sub_0203F19C (FieldSystem * fieldSystem, u16 param1)
+void FieldSystem_SetFlag (FieldSystem * fieldSystem, u16 flagID)
 {
-    VarsFlags_SetFlag(SaveData_GetVarsFlags(fieldSystem->saveData), param1);
-    return;
+    VarsFlags_SetFlag(SaveData_GetVarsFlags(fieldSystem->saveData), flagID);
 }
 
-void sub_0203F1B0 (FieldSystem * fieldSystem, u16 param1)
+void FieldSystem_ClearFlag (FieldSystem * fieldSystem, u16 flagID)
 {
-    VarsFlags_ClearFlag(SaveData_GetVarsFlags(fieldSystem->saveData), param1);
-    return;
+    VarsFlags_ClearFlag(SaveData_GetVarsFlags(fieldSystem->saveData), flagID);
 }
 
 void FieldSystem_ClearLocalFlags (FieldSystem * fieldSystem)
 {
-    int v0;
-    VarsFlags * v1;
-
-    v1 = SaveData_GetVarsFlags(fieldSystem->saveData);
-
-    memset(VarsFlags_GetFlagChunk(v1, 1), 0, (64 / 8));
-    memset(VarsFlags_GetVarAddress(v1, (0 + 0x4000)), 0, 2 * 32);
+    VarsFlags *varsFlags = SaveData_GetVarsFlags(fieldSystem->saveData);
 
-    return;
+    memset(VarsFlags_GetFlagChunk(varsFlags, 1), 0, (64 / 8));
+    memset(VarsFlags_GetVarAddress(varsFlags, (0 + VARS_START)), 0, 2 * 32);
 }
 
 void sub_0203F1FC (FieldSystem * fieldSystem)
 {
-    VarsFlags * v0;
-
-    v0 = SaveData_GetVarsFlags(fieldSystem->saveData);
-    memset(VarsFlags_GetFlagChunk(v0, 2400 + 320), 0, 192 / 8);
+    VarsFlags *varsFlags = SaveData_GetVarsFlags(fieldSystem->saveData);
 
-    return;
+    memset(VarsFlags_GetFlagChunk(varsFlags, 2400 + 320), 0, 192 / 8);
 }
 
-void sub_0203F21C (FieldSystem * fieldSystem, u16 param1, u16 param2, u16 param3, u16 param4)
+void FieldSystem_SetScriptParameters (FieldSystem * fieldSystem, u16 scriptParam0, u16 scriptParam1, u16 scriptParam2, u16 scriptParam3)
 {
-    *(u16 *)sub_0203F098(fieldSystem, 41) = param1;
-    *(u16 *)sub_0203F098(fieldSystem, 42) = param2;
-    *(u16 *)sub_0203F098(fieldSystem, 43) = param3;
-    *(u16 *)sub_0203F098(fieldSystem, 44) = param4;
+    *(u16 *)FieldSystem_GetScriptMember(fieldSystem, SCRIPT_DATA_PARAMETER_0) = scriptParam0;
+    *(u16 *)FieldSystem_GetScriptMember(fieldSystem, SCRIPT_DATA_PARAMETER_1) = scriptParam1;
+    *(u16 *)FieldSystem_GetScriptMember(fieldSystem, SCRIPT_DATA_PARAMETER_2) = scriptParam2;
+    *(u16 *)FieldSystem_GetScriptMember(fieldSystem, SCRIPT_DATA_PARAMETER_3) = scriptParam3;
 }
 
-u16 Script_GetTrainerID(u16 eventID)
+u16 Script_GetTrainerID(u16 scriptID)
 {
-    if (eventID < DOUBLE_BATTLE_EVENT_ID_OFFSET) {
-        return eventID - SINGLE_BATTLE_EVENT_ID_OFFSET + 1;
+    if (scriptID < SCRIPT_ID_OFFSET_DOUBLE_BATTLES) {
+        return scriptID - SCRIPT_ID_OFFSET_SINGLE_BATTLES + 1;
     } else {
-        return eventID - DOUBLE_BATTLE_EVENT_ID_OFFSET + 1;
+        return scriptID - SCRIPT_ID_OFFSET_DOUBLE_BATTLES + 1;
     }
 }
 
-BOOL sub_0203F278 (u16 param0)
+BOOL Script_GetDoubleBattleTrainerIndex (u16 scriptID)
```

Comment:
**suggestion:** Doesn't this routine check if a script refers to a single battle? In-context, at least, the call pattern seems to be to invoke this against a trainer's script.

Reply:
I can document the relevant script command that uses this soon, but it looks to me like it determines the trainer's "order" in a double battle based on what their trainer script ID is. Maybe this should be `Order` instead of `Index`?

Also returning a `BOOL` here is kind of weird - maybe it's worth changing it to return an `int`? It does match.

Reply:
I'd suggest that it returns an `int`, then, and probably the name `GetTrainerBattlerIndex` to implicitly tie it to the `BattleContext` fields.